features/api-layer/src/main/java/org/opennms/features/apilayer/utils/ModelMappers.java

```diff
@@ -58,11 +61,28 @@
 public class ModelMappers {
 
     public static Alarm toAlarm(OnmsAlarm alarm) {
-        return new AlarmBean(alarm);
+        return alarm == null ? null : ImmutableAlarm.newBuilder()
+                .setReductionKey(alarm.getReductionKey())
+                .setId(alarm.getId())
+                .setNode(toNode(alarm.getNode()))
+                .setManagedObjectInstance(alarm.getManagedObjectInstance())
+                .setManagedObjectType(alarm.getManagedObjectType())
+                .setAttributes(alarm.getDetails())
+                .setSeverity(toSeverity(alarm.getSeverity()))
+                .setRelatedAlarms(alarm.getRelatedAlarms()
+                        .stream()
+                        .map(ModelMappers::toAlarm)
+                        .collect(Collectors.toList()))
+                .setLogMessage(alarm.getLogMsg())
+                .setDescription(alarm.getDescription())
+                .setLastEventTime(alarm.getLastEventTime())
+                .setFirstEventTime(alarm.getFirstEventTime())
+                .setLastEvent(toEvent(alarm.getLastEvent()))
+                .build();
     }
 
     public static InMemoryEvent toEvent(Event event) {
-        return new InMemoryEventBean(event);
+        return event == null ? null : Mappers.getMapper(InMemoryEventMapper.class).map(event);
```

Comment:
Mappers can be stored in `static final` variables instead of rebuilding these each time.